src/SimpleGraphs/simpledigraph.jl

```diff
@@ -82,6 +82,146 @@ function SimpleDiGraph(g::AbstractSimpleGraph)
     return h
 end
 
+
+@inbounds function cleanupedges!(fadjlist::Vector{Vector{T}},
+                                 badjlist::Vector{Vector{T}}) where T<:Integer
+    neg = 0
+    for v in 1:length(fadjlist)
+        if !issorted(fadjlist[v])
+            sort!(fadjlist[v])
+        end
+        if !issorted(badjlist[v])
+            sort!(badjlist[v])
+        end
+        unique!(fadjlist[v])
+        unique!(badjlist[v])
+        neg += length(fadjlist[v])
+    end
+    return neg
+end
+
+function SimpleDiGraph(edge_list::Vector{SimpleDiGraphEdge{T}}) where T<:Integer
+    nvg = zero(T)
+    @inbounds(
+    for e in edge_list
+        nvg = max(nvg, src(e), dst(e)) 
+    end)
+   
+    list_sizes_out = ones(Int, nvg)
+    list_sizes_in = ones(Int, nvg)
+    degs_out = zeros(Int, nvg)
+    degs_in = zeros(Int, nvg)
+    @inbounds(
+    for e in edge_list
+        s, d = src(e), dst(e)
+        (s >= 1 && d >= 1) || continue
+        degs_out[s] += 1
+        degs_in[d] += 1
+    end)
+    
+    fadjlist = Vector{Vector{T}}(undef, nvg)
+    badjlist = Vector{Vector{T}}(undef, nvg)
+    @inbounds(
+    for v in 1:nvg
+        fadjlist[v] = Vector{T}(undef, degs_out[v])
+        badjlist[v] = Vector{T}(undef, degs_in[v])
+    end)
+
+    @inbounds(
+    for e in edge_list
+        s, d = src(e), dst(e)
+        (s >= 1 && d >= 1) || continue
+        fadjlist[s][list_sizes_out[s]] = d 
+        list_sizes_out[s] += 1
+        badjlist[d][list_sizes_in[d]] = s 
+        list_sizes_in[d] += 1
+    end)
+
+    g = SimpleDiGraph{T}()
+    g.fadjlist = fadjlist
+    g.badjlist = badjlist
+    g.ne = cleanupedges!(fadjlist, badjlist)
```

Comment:
You're calling `cleanupedges!` here, which sorts and dedups fadjlist and badjlist, but you've already assigned them to `g`. The issue is that the adjacency lists of a SimpleGraph are presumed to be sorted and deduplicated. So if they're unsorted/redundant at the time you assign them to `g`, then that's wrong. If they're sorted/unique, then you don't need to re-sort. Since this is likely modifying the backing array (and changing `g.fadjlist` which is a reference to it), it should probably come first just for clarity.